src/System.Linq/src/System/Linq/Aggregate.cs

```diff
@@ -39,41 +32,14 @@ public static TSource Aggregate<TSource>(this IEnumerable<TSource> source, Func<
 
         public static TAccumulate Aggregate<TSource, TAccumulate>(this IEnumerable<TSource> source, TAccumulate seed, Func<TAccumulate, TSource, TAccumulate> func)
         {
-            if (source == null)
-            {
-                throw Error.ArgumentNull(nameof(source));
-            }
-
-            if (func == null)
-            {
-                throw Error.ArgumentNull(nameof(func));
-            }
-
-            TAccumulate result = seed;
-            foreach (TSource element in source)
-            {
-                result = func(result, element);
-            }
-
-            return result;
+            return source.Aggregate(seed, func, result => result);
```

Comment:
True, but is removing duplicated code. it makes it more maintainable. Also this extra delegate is called only once per query independent of size of the array. So, the complexity only increased by O(1)